From 53e2ce4160c61fc239d1dc3a6fb22f2b8edbaf3b Mon Sep 17 00:00:00 2001
From: Alan Zhao <ayzhao@google.com>
Date: Thu, 22 Sep 2022 23:46:21 -0700
Subject: [PATCH 1/2] Disable several clang warnings

These silences are required to get 7z to build. The alternative, which
is to fix the code, is unwieldy because these errors trigger in so many
places that the resulting patch files would become very unmanagable.
Additionally, the 7z source code doesn't have a Git repository, and
there are no known ways to contribute fixes upstream.
---
 CPP/7zip/warn_clang.mak | 9 +++++++++
 CPP/7zip/warn_gcc.mak   | 3 +--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/CPP/7zip/warn_clang.mak b/CPP/7zip/warn_clang.mak
index a299fbc..44c08e7 100755
--- a/CPP/7zip/warn_clang.mak
+++ b/CPP/7zip/warn_clang.mak
@@ -30,6 +30,15 @@ CFLAGS_WARN_CLANG_12= $(CFLAGS_WARN_CLANG_3_8) \
 
 CFLAGS_WARN_1 = \
   -Wno-deprecated-copy-dtor \
+  -Wno-unknown-warning-option \
+  -Wno-reserved-identifier \
+  -Wno-suggest-override \
+  -Wno-suggest-destructor-override \
+  -Wno-declaration-after-statement \
+  -Wno-unused-but-set-variable \
+  -Wno-poison-system-directories \
+  -Wno-unguarded-availability-new \
+  -Wno-c++11-extensions \
 
 
 
diff --git a/CPP/7zip/warn_gcc.mak b/CPP/7zip/warn_gcc.mak
index afc0c9d..de7e2c9 100755
--- a/CPP/7zip/warn_gcc.mak
+++ b/CPP/7zip/warn_gcc.mak
@@ -19,7 +19,6 @@ CFLAGS_WARN_GCC_6 = \
 
 CFLAGS_WARN_GCC_9 = \
   -Waddress \
-  -Waddress-of-packed-member \
   -Waggressive-loop-optimizations \
   -Wattributes \
   -Wbool-compare \
@@ -54,4 +53,4 @@ CFLAGS_WARN = $(CFLAGS_WARN_GCC_9) \
 
 #  $(CFLAGS_WARN_GCC_PPMD_UNALIGNED)
 
-  
\ No newline at end of file
+  
-- 
2.37.3.998.g577e59143f-goog

